Socket improvments

palaviv opened the pull request:
- socket.recv read amount of bytes asked
- Add validation to connect and bind address

rickyhan commented:
There are some linter warnings from your commit:

```
   --> vm/src/stdlib/socket.rs:182:43
    |
182 | fn socket_listen(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
    |                                           ^^^^ help: consider prefixing with an underscore: `_args`
    |
    = note: #[warn(unused_variables)] on by default

warning: unused variable: `bufsize`
   --> vm/src/stdlib/socket.rs:226:36
    |
226 |         required = [(zelf, None), (bufsize, Some(vm.ctx.int_type()))]
    |                                    ^^^^^^^ help: consider prefixing with an underscore: `_bufsize`

warning: variant is never constructed: `UdpSocket`
  --> vm/src/stdlib/socket.rs:53:5
   |
53 |     UdpSocket(UdpSocket),
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(dead_code)] on by default

```

palaviv commented:
@rickyhan I fixed the warnings.
I also added `getsockname` so I can pass 0 as port in the tests. I think that they failed because the port was on TIME_WAIT.
